### PR TITLE
If the file already exists we need to grant group read permissions ex…

### PR DESCRIPTION
**Pull request by vinayakankugoyal**
…plicitly. 

**What type of PR is this?**

Add one of the following kinds:
/kind bug

**What this PR does / why we need it**:
If the file passed into writ-pki-data already exists and had permissions 700, even if we write to it with umask 027 and chgrp the group does not get read permissions. We need to explicitly grant the group read permissions.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by PurelyApplied** [comment]
\<heavy sigh\>

\ok-to-test

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -622,6 +622,7 @@ function write-pki-data {
   if [[ -n "${KUBE_PKI_READERS_GROUP:-}" ]]; then
     (umask 027; echo "${data}" | base64 --decode > "${path}")
     chgrp "${KUBE_PKI_READERS_GROUP:-}" "${path}"
+    chmod g+r "${path}"
```
Comment (roycaihw) [inline cluster/gce/gci/configure-helper.sh:625]: When will these files be created already? Do we know their permissions are always 700? Does this issue happen to `umask 077` below?

Comment (vinayakankugoyal) [reply cluster/gce/gci/configure-helper.sh:625]: In gke if you upgrade the master these files already exist. It happens if you upgrade from a version that did not have PR https://github.com/kubernetes/kubernetes/pull/95418 to a version where you had it. This bug will not affect components that run as root only components that run as non-root on the gke-control-plane and need to access these secrets which is exactly one component. 
When the code with umask 077 gets run no component is running as non root (all are running as root) so they are unaffected.

Comment (roycaihw) [reply cluster/gce/gci/configure-helper.sh:625]: If the master is downgraded from umask 027 to 077, are we concerned about leaking read access to certain group?

Comment (vinayakankugoyal) [reply cluster/gce/gci/configure-helper.sh:625]: To join a group you need root access, so to join that group an attacker would need root access at which point they already have access to everything. So I don't think it is a concern.

Comment (roycaihw) [reply cluster/gce/gci/configure-helper.sh:625]: SGTM. Thanks

**Pull request merged**